namer/consul/src/main/scala/io/buoyant/namer/consul/InstrumentedWatch.scala

```diff
@@ -0,0 +1,54 @@
+package io.buoyant.namer.consul
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.http
+import com.twitter.util.{Return, Throw, Time, Try, Future}
+import io.buoyant.consul.v1
+import scala.collection.mutable
+
+/**
+ * PollState holds metadata about the calls on consul api.
+ * This class is intended to be serialized.
+ */
+class PollState[Req, Rep] {
+
+  @JsonIgnore
+  def recordApiCall(req: Req): Unit = synchronized {
+    request = Some(req)
+    lastRequestAt = Some(Time.now.toString)
+  }
+
+  @JsonIgnore
+  def recordResponse(rep: Try[Rep]): Unit = synchronized {
+    lastResponseAt = Some(Time.now.toString)
+    rep match {
+      case Return(r) =>
+        response = Some(r)
+        error = None
+      case Throw(e) =>
+        error = Some(e)
+        response = None
+    }
+  }
+
+  // These fields exist to be serialized.
+  protected var request: Option[Req] = None
+  protected var lastRequestAt: Option[String] = None
+  protected var response: Option[Rep] = None
+  protected var lastResponseAt: Option[String] = None
+  protected var error: Option[Throwable] = None
```

Comment:
I just noticed this in the consul dtab store PR but it applies here too:

arbitrary throwables cannot be serialized. Let's make this an Option[String] and in the Throw case of recordResponse, we should populate this with the throwable's message.

Reply:
I'll put myself to it after dinner